scripts/scil_visualize_bundles_mosaic.py

```diff
@@ -200,13 +224,28 @@ def main():
                                     i + text_pos_x, j + text_pos_y, font)
 
         else:
+            if args.uniform_coloring:
+                colors = args.uniform_coloring
+            elif args.random_coloring is not None:
+                colors = random_rgb()
             # Select the streamlines to plot
-            bundle_tractogram_file = nib.streamlines.load(bundle_file)
-            streamlines = bundle_tractogram_file.streamlines
-
-            tubes = actor.line(streamlines)
-
-            number_streamlines = len(streamlines)
+            if bundle_ext in ['.tck', '.trk']:
+                if args.random_coloring is None and args.uniform_coloring is None:
+                    colors = None
+                bundle_tractogram_file = nib.streamlines.load(bundle_file)
```

Comment:
why not load_with_ref ? or at least check compatibility between anat and bundles.